README.md

```diff
@@ -1011,6 +1012,34 @@ std::vector<std::uint8_t> v_ubjson = json::to_ubjson(j);
 json j_from_ubjson = json::from_ubjson(v_ubjson);
 ```
 
+### User self-defined cases
+
+This JSON library is very flexible in usage, you can customize and implement you desired functions in your own projects, and can also self-defined data type that you want to parse, or handle some edge cases by yourselves.
+
+#### Custom comparison function
+
+As we know, the result of comparing two floating point numbers may be uncertain, which
+depends on the system architecture, the compiler, and the absolute/relative error value we can acccpt.
+For this reason, you can self-defined function according to specific scenarios, like
+
+```cpp
+bool is_equal(const_reference lhs, const_reference rhs)
```

Comment:
I am not sure whether the added section to the README actually adds value, as the [documentation of `operator==`](https://nlohmann.github.io/json/classnlohmann_1_1basic__json_a122640e7e2db1814fc7bbb3c122ec76e.html#a122640e7e2db1814fc7bbb3c122ec76e) already describes the current behavior and provides and example for a different comparison.

That said, I think the code snippet is better suited in the documentation of `operator==` while letting the README unchanged.

Reply:
> That said, I think the code snippet is better suited in the documentation of operator== while letting the README unchanged.

Fine. I will move it.